docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,346 @@
+    if (metadataReader.GetGuid(customDebugInformation.Kind) == metadataReferencesGuid)
+    {
+        var blobReader = metadataReader.GetBlobReader(customDebugInformation.Value);
+        
+        // Each loop is one reference
+        while (blobReader.RemainingBytes > 0)
+        {
+            // Order of information
+            // File name (null terminated string): A.exe
+            // Extern Alias (null terminated string): a1,a2,a3
+            // EmbedInteropTypes/MetadataImageKind (byte)
+            // COFF header Timestamp field (4 byte int)
+            // COFF header SizeOfImage field (4 byte int)
+            // MVID (Guid, 24 bytes)
+
+            var terminatorIndex = blobReader.IndexOf(0);
+            Assert.NotEqual(-1, terminatorIndex);
+
+            var name = blobReader.ReadUTF8(terminatorIndex);
+
+            // Skip the null terminator
+            blobReader.ReadByte();
+
+            terminatorIndex = blobReader.IndexOf(0);
+            Assert.NotEqual(-1, terminatorIndex);
+
+            var externAliases = blobReader.ReadUTF8(terminatorIndex);
+
+            // Skip the null terminator
+            blobReader.ReadByte();
+
+            var embedInteropTypesAndKind = blobReader.ReadByte();
+            var embedInteropTypes = (embedInteropTypesAndKind & 0b10) == 0b10;
+            var kind = (embedInteropTypesAndKind & 0b1) == 0b1
+                ? MetadataImageKind.Assembly
+                : MetadataImageKind.Module;
+
+            var timestamp = blobReader.ReadInt32();
+            var imageSize = blobReader.ReadInt32();
+            var mvid = blobReader.ReadGuid();
+
+            Console.WriteLine(name);
+            Console.WriteLine($"Extern Aliases: \"{externAliases}\"");
+            Console.WriteLine($"Embed Interop Types: {embedInteropTypes}");
+            Console.WriteLine($"Metadata Image Kind: {kind}");
+            Console.WriteLine($"Timestamp: {timestamp}");
+            Console.WriteLine($"Image Size: {imageSize}");
+            Console.WriteLine($"MVID: {mvid}");
+            Console.WriteLine();
+        }
+    }
+}
+```
+
+#### Compiler Flag Key Value Pairs
+
+The remaining values will be stored as key value pairs in the pdb. The storage format will be UTF8 encoded key value pairs that are null terminated. Order is not guaranteed. Any values left out can be assumed to be the default for the type. Keys may be different for Visual Basic and CSharp. They are serialized to reflect the command line arguments representing the same values
+
+Example: 
+
+`compilerversion\01.2.3-example-sha\0sourceencoding\0utf-8\0checked\01\0unsafe\01\0langversion\0latest\0nullable\0Enable` 
+
+##### Options For CSharp
+
+See [compiler options](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/listed-alphabetically) documentation
+
+* "[appconfig](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/appconfig-compiler-option)"
```

Comment:
>appconfig [](start = 4, length = 9)

This is somewhat confusing - the key in the PDB is called `portability-policy` but here you mention `appconfig`, which is the name of the command line switch. On the other hand, "compilerversion" below is the PDB key. I think it would be good to clearly differentiate between these.
Perhaps make this a table with first column being the PDB key, the second corresponding command line switch and third description.

Reply:
Made a table, think it is much clearer. Linked complex topics to other sections in the document.